src/libsync/vfs/cfapi/cfapiwrapper.cpp

```diff
@@ -124,21 +124,75 @@ void CALLBACK cfApiFetchDataCallback(const CF_CALLBACK_INFO *callbackInfo, const
     }
 
     qint64 offset = 0;
+    QByteArray protrudingData;
+    bool protrudingDataLeft = false;
 
     QObject::connect(&socket, &QLocalSocket::readyRead, &loop, [&] {
-        auto data = socket.readAll();
+        const auto data = socket.readAll();
         if (data.isEmpty()) {
             qCWarning(lcCfApiWrapper) << "Unexpected empty data received" << requestId;
             sendTransferError();
             loop.quit();
             return;
         }
-        sendTransferInfo(data, offset);
-        offset += data.length();
+
+        QByteArray dataToSent;
+        // CFAPI expectes sended blocks to be of a specified size.
+        // Only the last sended block is allowed to be smaller than the specified block size
+        // and therefore marks the end of the transmission.
+        static constexpr auto CFAPI_BLOCK_SIZE = 4096;
+
+        int dataIndex = 0;
+        // Is there any data left from last run?
+        if (!protrudingData.isEmpty()) {
+            QByteArray dataToSent;
+            dataToSent.append(protrudingData);
+
+            // Fill up remaining first block if possible
+            const auto fillupSize = CFAPI_BLOCK_SIZE - dataToSent.size();
```

Comment:
@FlexW Now, I am wondering. Can we optimize this logic a bit, by allowing multiples of `CFAPI_BLOCK_SIZE ` will it work if the block size is **8192** or **16384**? As far as I remember, the block size is not required to be strictly 4096, but, any multiple of 4096 is allowed too?

Reply:
I tried that in the first place but it didn't worked

Reply:
> I tried that in the first place but it didn't worked

I am quite surprised since it works for me. I have updated my branch if you are curious, and it works with chunks bigger than 4096 as long as they are multiples of 4096. Even 12288-sized blocks work fine for me.

Which error are you getting? Let's try to investigate it.

Reply:
Resolve that conversation as we discussed it in the chat